tracer/src/Datadog.Trace/Ci/IntelligentTestRunnerClient.cs

```diff
@@ -730,11 +693,25 @@ private async Task<ObjectPackFilesResult> GetObjectsPackFileFromWorkingDirectory
                     Log.Debug(exceptionDispatchInfo.SourceException, "Unable to communicate with the server");
                 }
 
-                // Execute retry delay
-                await Task.Delay(sleepDuration).ConfigureAwait(false);
-                retryCount++;
-                sleepDuration *= 2;
+                if (sourceException is RateLimitException rlException)
+                {
+                    if (rlException?.DelayTimeInSeconds is { } delayTimeInSeconds)
+                    {
+                        await Task.Delay(TimeSpan.FromSeconds(delayTimeInSeconds)).ConfigureAwait(false);
+                    }
+                    else
+                    {
+                        exceptionDispatchInfo.Throw();
+                    }
+                }
```

Comment:
If you go with my suggestion above, you can simplify this
```suggestion
                if (sourceException is RateLimitException { DelayTimeInSeconds : { } delayTimeInSeconds})
                {
                    await Task.Delay(TimeSpan.FromSeconds(delayTimeInSeconds)).ConfigureAwait(false);
                }
```

Reply:
done in [b4dd46d](https://github.com/DataDog/dd-trace-dotnet/pull/3490/commits/b4dd46dabd3c99f8047cbb92342899da6df8f9e4)